test/Elastic.Apm.AspNetCore.Tests/BodyCapturingTests.cs

```diff
@@ -30,77 +34,258 @@ namespace Elastic.Apm.AspNetCore.Tests
+
 			// build test data, which we send to the sample app
-			var data = new BaseReportFilter<SendMessageFilter>();
-			data.ReportFilter = new SendMessageFilter();
-			data.ReportFilter.Body = "message body";
-			data.ReportFilter.SenderApplicationCode = "26";
-			data.ReportFilter.MediaType = "TokenBasedSms";
-			data.ReportFilter.Recipients = new List<string> { "abc123" };
+			var data = new BaseReportFilter<SendMessageFilter>
+			{
+				ReportFilter = new SendMessageFilter
+				{
+					Body = "message body",
+					SenderApplicationCode = "26",
+					MediaType = "TokenBasedSms",
+					Recipients = new List<string> { "abc123" }
+				}
+			};
 
 			var body = JsonConvert.SerializeObject(data,
 				new JsonSerializerSettings { ContractResolver = new CamelCasePropertyNamesContractResolver() });
 
 			// send data to the sample app
-			var httpClient = new HttpClient();
-			httpClient.BaseAddress = new Uri("http://localhost:5903");
-			var result = await httpClient.PostAsync("api/Home/Send", new StringContent(body, Encoding.UTF8, "application/json"));
+			var result = await sutEnv.HttpClient.PostAsync("api/Home/Send", new StringContent(body, Encoding.UTF8, "application/json"));
 
 			// make sure the sample app received the data
 			result.StatusCode.Should().Be(200);
 
 			// and make sure the data is captured by the agent
-			_payloadSender1.FirstTransaction.Should().NotBeNull();
-			_payloadSender1.FirstTransaction.Context.Request.Body.Should().Be(body);
+			sutEnv.MockPayloadSender.FirstTransaction.Should().NotBeNull();
+			sutEnv.MockPayloadSender.FirstTransaction.Context.Request.Body.Should().Be(body);
 		}
 
-		public Task InitializeAsync()
+		private static IEnumerable<OptionsTestVariant> BuildOptionsTestVariants()
 		{
-			_agent = new ApmAgent(new TestAgentComponents(payloadSender: _payloadSender1,
-				config: new MockConfigSnapshot(new NoopLogger(), captureBody: ConfigConsts.SupportedValues.CaptureBodyAll)));
+			var captureBodyContentTypesVariants = new[]
+			{
+				ConfigConsts.DefaultValues.CaptureBodyContentTypes,
+				ConfigConsts.DefaultValues.CaptureBodyContentTypes + ", " + MyCustomContentType
+			};
 
-			_taskForSampleApp = Program.CreateWebHostBuilder(null)
-				.ConfigureServices(services =>
+			foreach (var captureBody in ConfigConsts.SupportedValues.CaptureBodySupportedValues)
+			{
+				foreach (var captureBodyContentTypes in captureBodyContentTypesVariants)
+				{
+					foreach (var isSampled in new[] { false, true})
 					{
-						Startup.ConfigureServicesExceptMvc(services);
+						yield return new OptionsTestVariant
+						{
+							CaptureBody = captureBody,
+							CaptureBodyContentTypes = captureBodyContentTypes,
+							IsSampled = isSampled
+						};
+					}
+				}
+			}
+		}
+
+		[Theory]
+		[MemberData(nameof(OptionsChangedAfterStartTestVariants))]
+		public async Task Options_changed_after_start(int startCfgVariantIndex, OptionsTestVariant startCfgVariant)
```

Comment:
It'd be really nice to add some comments to this one. To be honest, I already started going through this PR before EAH and this was the point where I gave up - some comments would really help here.

Reply:
Done.